src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/UnconfiguredProjectContextProvider.cs

```diff
@@ -181,8 +181,8 @@ private bool TryGetConfiguredProjectState(ConfiguredProject configuredProject, o
                 }
                 else
                 {
-                    workspaceProjectContext = null;
-                    configuredProjectHostObject = null;
+                    workspaceProjectContext = null!;
```

Comment:
I decided to do this rather than annotate them, as we don't have the attributes to annotate `Try` methods properly yet.

Reply:
We could add it easily right?

```C#
namespace System.Runtime.CompilerServices
{
    [AttributeUsage(AttributeTargets.Parameter,
                   AllowMultiple = false)]
    public class NotNullWhenTrueAttribute : Attribute
    {
        public NotNullWhenTrueAttribute() { }
    }
}
```

---
In reply to: [242257953](https://github.com/dotnet/project-system/pull/4404#discussion_r242257953) [](ancestors = 242257953)

Reply:
[https://github.com/dotnet/roslyn/blob/master//src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs#L55-L64](https://github.com/dotnet/roslyn/blob/master//src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs#L55-L64)

---
In reply to: [242265057](https://github.com/dotnet/project-system/pull/4404#discussion_r242265057) [](ancestors = 242265057,242257953)

Reply:
Even having it still seems to require `!`

Reply:
It should work if you also update the parameters to be nullable.

```C#
private bool TryGetConfiguredProjectState(ConfiguredProject configuredProject, 
									      [NotNullWhenTrue] out IWorkspaceProjectContext? workspaceProjectContext,
                                          [NotNullWhenTrue] out IConfiguredProjectHostObject? configuredProjectHostObject)
{
	...
	workspaceProjectContext = null;
	configuredProjectHostObject = null;
	return false;
}
```

That way the compiler _should_ not warn at the call-site if you check for `true` but _will_ warn if you do not

Reply:
Ok, done.